edge-hub/src/Microsoft.Azure.Devices.Routing.Core/checkpointers/Checkpointer.cs

```diff
@@ -32,10 +32,40 @@ public class Checkpointer : ICheckpointer
             this.UnhealthySince = checkpointData.UnhealthySince;
             this.Proposed = checkpointData.Offset;
             this.closed = new AtomicBoolean(false);
+
+            // The endpoint ID and priority is encoded into the checkpointer ID
+            // using the following format:
+            //   {endpointId}_Pri{priority}
+            // We use "_Pri" as a delimiter to parse the endpoint ID and priority
+            // back out for metrics reporting
+            string[] tokens = id.Split(new string[] { "_Pri" }, System.StringSplitOptions.RemoveEmptyEntries);
```

Comment:
If a user enters an endpoint with a name "foo_Pribar" then the metrics won't get reported. Not a likely case, but something to keep in mind

Reply:
Yea, to really fix that would involve a lot more plumbing though.  Let's come back in a different PR to address it.

6607970 - Plumb endpoint ID and priority to Checkpointer for accurate metrics reporting